velox/connectors/Connector.h

```diff
@@ -304,7 +308,8 @@ class Connector {
       const std::unordered_map<
           std::string,
           std::shared_ptr<connector::ColumnHandle>>& columnHandles,
-      ConnectorQueryCtx* FOLLY_NONNULL connectorQueryCtx) = 0;
+      ConnectorQueryCtx* FOLLY_NONNULL connectorQueryCtx,
+      const dwio::common::ReaderOptions& options) = 0;
```

Comment:
I cannot find the definitive source of that config object that comes from the "consumer" of Velox.  I looked at gluten source and there appears to be no path to get these values into a config that ends up in HiveConnector's CTX (too many configs and CTXs and code paths to track). In fact the ONLY way gluten is passing in config values is by directly setting "FLAGS_<value_name>" from Velox.  At this point, I am not sure what to do. I think my first solution to this problem was the correct solution. I think adding the "max_coalesce_distance_bytes" should be simply a new FLAG_ value exposed as is.

Reply:
There should be a way to pass the config into connector, maybe check with @jinchengchenghh on https://github.com/facebookincubator/velox/pull/4040, where she pass `file_column_names_read_as_lower_case` as config into the connector

Reply:
Ok, I will first rebase the conflict then re-implement the option addition. Make take a day or 2.

Reply:
Ok, I think I have finished the changes and they passed in CI.

Reply:
The failure in CI appears to be an unstable test since I did not modify the memory code at all. I have seen it before along with another one in the SharedArbitration.concurrency... tests.